mx_fetchjdk.py

```diff
@@ -0,0 +1,197 @@
+        untar = mx.TarExtractor(archive_location)
+
+        if not args["quiet"]:
+            print("Installing...")
+
+        with untar._open() as tar_file:
+            curr_full_jdk_path = untar._getnames(tar_file)[0]
+
+        try:
+            untar.extract(jdk_path)
+        except:
+            os.remove(join(jdk_path, jdk_archive))
+            shutil.rmtree(curr_full_jdk_path)
+            mx.abort("Error parsing archive. Please try again")
+        if not args["keep-archive"]:
+            os.remove(join(jdk_path, jdk_archive))
+        os.rename(join(jdk_path, curr_full_jdk_path), full_jdk_path)
+
+    elif not args["quiet"]:
+        mx.warn("Requested JDK was already present")
+
+    if mx.is_darwin() and exists(join(full_jdk_path, 'Contents', 'Home')):
+        if args["strip-contents-home"]:
+            tmp_full_jdk_path = full_jdk_path + ".tmp"
+            shutil.move(full_jdk_path, tmp_full_jdk_path)
+            shutil.move(join(tmp_full_jdk_path, 'Contents', 'Home'), full_jdk_path)
+            shutil.rmtree(tmp_full_jdk_path)
+        else:
+            full_jdk_path = join(full_jdk_path, 'Contents', 'Home')
+
+    if "alias" in args:
+        alias_full_path = join(jdk_path, args["alias"])
+        if exists(alias_full_path) or islink(alias_full_path):
+            if isdir(alias_full_path) and not islink(alias_full_path):
+                shutil.rmtree(alias_full_path)
+            else:
+                os.remove(alias_full_path)
+
+        if not (mx.is_windows() or mx.is_cygwin()):
+            os.symlink(full_jdk_path, alias_full_path)
+        else:
+            copytree(full_jdk_path, alias_full_path, symlinks=True) # fallback for windows
+        full_jdk_path = alias_full_path
+
+    if not args["quiet"]:
+        print("Run the following to set JAVA_HOME in your shell:")
+
+    print('export JAVA_HOME={}'.format(abspath(full_jdk_path)))
+
+
+def _parse_fetchjdk_settings(args):
+    settings = {}
+    settings["quiet"] = False
+    settings["keep-archive"] = False
+    settings["jdk-path"] = abspath(join('bin', 'jdks'))
+
+    jdk_paths = find_system_jdks()
+    if len(jdk_paths) > 0:
+        settings["jdk-path"] = next(iter(jdk_paths))
+        found_jdk_path = True
+    else:
+        found_jdk_path = False
+
+    common_location = join(_mx_home, 'common.json')
+
+    parser = ArgumentParser(prog='mx fetch-jdk')
+    parser.add_argument('--java-distribution', action='store', help='JDK distribution that should be downloaded (e.g., "labsjdk-ce-11" or "openjdk8")')
+    parser.add_argument('--configuration', action='store', help='location of configuration json file (default: \'{}\')'.format(common_location))
+    parser.add_argument('--to', action='store', help='location where JDK would be downloaded (default: \'{}\')'.format(settings["jdk-path"]))
```

Comment:
As mentioned above, the `--to` argument is for specifying the directory in which the JDK is to be installed, not the `JAVA_HOME` directory of the installed JDK.

Reply:
Fixed